src/Nixfmt/Util.hs

```diff
@@ -0,0 +1,52 @@
+module Nixfmt.Util
+    ( manyList
+    , someList
+    , manyP
+    , someP
+    , manyText
+    , someText
+    , commonPrefix
+    , dropCommonIndentation
+    ) where
+
+import           Data.Char
+import           Data.Maybe
+import           Data.Text       as Text hiding (map, takeWhile)
+import           Data.Void
+import           Text.Megaparsec hiding (empty)
+
+type Parser = Parsec Void Text
```

Comment:
Do you really need to fix the stream type for your utils? They all seem to be generic enough.